src/api/route-services/access/index.js

```diff
@@ -0,0 +1,100 @@
+
+const _ = require('lodash');
+
+const config = require('../../../config');
+const AWS = require('../../../utils/requireAWS');
+const getLogger = require('../../../utils/getLogger');
+
+const {
+  createDynamoDbInstance,
+  convertToJsObject,
+  convertToDynamoDbRecord,
+} = require('../../../utils/dynamoDb');
+
+const { isRoleAuthorized } = require('./roles');
+
+const logger = getLogger('[AccessService] - ');
+
+class AccessService {
+  constructor() {
+    this.userAccessTableName = `user-access-${config.clusterEnv}`;
+    this.inviteAccessTableName = `invite-access-${config.clusterEnv}`;
+  }
+
+  // grantRole adds or replaces role for userId and experimentId
+  async grantRole(userId, experimentId, projectId, role) {
+    logger.log(`PUT experiment role ${experimentId} data`);
+
+    const params = {
+      TableName: this.userAccessTableName,
+      Item: {
+        userId,
+        experimentId,
+        projectId,
+        role,
+      },
+    };
+
+    const docClient = new AWS.DynamoDB.DocumentClient({
+      region: config.awsRegion,
+    });
```

Comment:
this `DocumentClient` looks much more convenient than having to marshall and unmarshall data

Reply:
can we use it for other methods too?

Reply:
yes and no, I read that it has issues with paginations in some cases. considering we are going to remove this code (when we move to SQL) I did not spend more time making it consistent

Reply:
makes sense!